pulp_ansible/app/tasks/collections.py

```diff
@@ -378,12 +385,50 @@ def _build_url(path_or_url):
                         if result.get("version") and not download_url:
                             version_url = _build_url(result["href"])
                             not_done.update([remote.get_downloader(url=version_url).run()])
+                            additional_metadata[version_url] = {
+                                "docs_blob_url": f"{version_url}docs-blob/"
+                            }
 
                         if download_url:
+                            metadata = additional_metadata.get(_build_url(data["href"]), {})
+                            data["docs_blob_url"] = metadata.get("docs_blob_url")
                             yield data
                             progress_bar.increment()
 
 
+class DocsBlobDownloader(ArtifactDownloader):
+    """
+    Stage for downloading docs_blob.
+
+    Args:
+        max_concurrent_content (int): The maximum number of
+            :class:`~pulpcore.plugin.stages.DeclarativeContent` instances to handle simultaneously.
+            Default is 200.
+        args: unused positional arguments passed along to :class:`~pulpcore.plugin.stages.Stage`.
+        kwargs: unused keyword arguments passed along to :class:`~pulpcore.plugin.stages.Stage`.
+    """
+
+    async def _handle_content_unit(self, d_content):
+        """Handle one content unit.
+
+        Returns:
+            The number of downloads
+        """
+        downloaders_for_content = []
+        for d_artifact in d_content.d_artifacts:
```

Comment:
The move of extradata to `DeclarativeContent` I believe will remove the need for this loop.

Reply:
I would need to get the `DeclarativeArtifact` the same way, as the downloader comes from there